docs/docs/upgrade-gatsby-and-dependencies.md

```diff
@@ -0,0 +1,142 @@
+
+## Why you should upgrade Gatsby and its dependencies
+
+Every new version of every package comes with improvements on multiple categories from performance, accesibility, to security and bug fixes, among others, so it is important to upgrade both Gatsby and its dependencies to get the latest improvements in every one of these categories.
+
+Constantly upgrading your dependencies on minor or patch releases also helps you to make major upgrades easier and to identify ossible soon to be deprecated functionality or APIs.
+
+## How to identify possible upgrades
+
+Depending on your current setup, you can use one of [yarn](https://yarnpkg.com/) or [npm](https://www.npmjs.com/get-npm) to update your dependencies.
+
+To start, you can run the outdated command to identify new releases for all your dependencies. Remember that you only have to use one of npm or yarn, not both of them.
+
+```shell
+npm outdated
+```
+
+```shell
+yarn outdated
+```
+
+This will output a table indicating which packages have new versions available and what is the latest version for each one.
+
+```
+Package                            Current   Wanted   Latest  Location
+gatsby                             2.15.13  2.15.13  2.15.20
+```
+
+## Configure your dependencies for upgrades
+
+Depending on whether you want to update Gatsby and its dependencies for minor or patch releases you need to modify your `package.json` accordingly. If you only want to update for patch releases, you can add a tilde before the version of your package:
+
+```title=package.json
+"dependencies"{
+  "gatsby": "~2.15.13",
+}
+```
+
+For both patch and minor updates, add a caret before the version of your package:
+
+```title=package.json
+"dependencies"{
+  "gatsby": "^2.15.13",
+}
+```
+
+For major updates follow up with the corresponding guide from the [Release and Migrations](https://www.gatsbyjs.org/docs/releases-and-migration/) reference guide overview.
+
+In case that you are updating Gatsby, be sure to also update Gatsby related plugins, you can identify them because their name starts with `gatsby-`, this only applies to plugins managed in the gatsbyjs/gatsby repo, for community plugins check beforehand if there is a new version you can upgrade to.
+
+## Updating all your dependencies at once
+
+After adding the corresponding annotations into your `package.json` file, you can run the update command in npm or the upgrade command if you are using yarn, remember that you only have to use one of yarn or npm:
+
+```shell
+npm update
+```
+
+```shell
+yarn upgrade
+```
+
+This will upgrade all your packages to the latest [wanted](https://docs.npmjs.com/cli/outdated) version, this means the latest patch in case that you specified that only patch updates were allowed, and so on for minor and major update annotations.
```

Comment:
```suggestion
This will upgrade all your packages to the latest [wanted version](https://docs.npmjs.com/cli/outdated): such as the latest patch, minor, or major update depending on your annotations in `package.json`.
```